rest_framework/renderers.py

```diff
@@ -342,8 +342,24 @@ def render_field(self, field, parent_style):
         # Get a clone of the field with text-only value representation.
         field = field.as_form_field()
 
-        if style.get('input_type') == 'datetime-local' and isinstance(field.value, str):
-            field.value = field.value.rstrip('Z')
+        if style.get('input_type') == 'datetime-local':
+            has_format = hasattr(field._field, "format")
+
+            if has_format:
+                if field._field.format is None:
+                    datetime_field_value = field.value
+                else:
+                    datetime_field_value = datetime.datetime.strptime(field.value, field._field.format)
+            else:
+                if api_settings.DATETIME_FORMAT == ISO_8601:
+                    datetime_field_value = datetime.datetime.fromisoformat(field.value.rstrip('Z'))
```

Comment:
This assumes that `field.value` is a string, although this change also removes the `isinstance` check in line 345. (Why) is this safe?

I'm not sure under which circumstances the expression would not be a string. The type check was introduced in https://github.com/mgaligniana/django-rest-framework/commit/df7b6fcf58417fd95e49655eb140b387899b1ceb#diff-d969c1c82f7a9d0c9b55cd44872d2ad719203180f1404e974e6daaf33be020ebR387.